In [1]:
import numpy as np
import statsmodels.formula.api as sm

In [2]:
np.random.seed(1)
x = np.random.randn(100)
y = 2*x+np.random.randn(100)

### A simple linear regression of _y onto x_, without an intercept

In [3]:
model_y_onto_x = sm.OLS(y,x)
results_y_onto_x = model_y_onto_x.fit()
results_y_onto_x.summary2()

<class 'statsmodels.iolib.summary2.Summary'>
"""
                 Results: Ordinary least squares
=================================================================
Model:              OLS              Adj. R-squared:     0.796   
Dependent Variable: y                AIC:                273.3438
Date:               2017-11-09 17:34 BIC:                275.9489
No. Observations:   100              Log-Likelihood:     -135.67 
Df Model:           1                F-statistic:        391.7   
Df Residuals:       99               Prob (F-statistic): 3.46e-36
R-squared:          0.798            Scale:              0.89191 
--------------------------------------------------------------------
         Coef.     Std.Err.       t       P>|t|     [0.025    0.975]
--------------------------------------------------------------------
x1       2.1067      0.1064    19.7918    0.0000    1.8955    2.3180
-----------------------------------------------------------------
Omnibus:               0.880        Durbin-Watson:          2.106
Prob(Omnibus):         0.644        Jarque-Bera (JB):       0.554
Skew:                  -0.172       Prob(JB):               0.758
Kurtosis:              3.119        Condition No.:          1    
=================================================================

"""

Coefficient estimate for ß is 2.1067 with a standard error of 0.1064. That gives a t-statistic of 19.7918 and a p-value of less than 0.0001 associated with the null hypothesis H₀ : ß = 0, so we can reject the null hypothesis.

### A simple linear regression of _x onto y_, without an intercept

In [4]:
model_x_onto_y = sm.OLS(x,y)
results_x_onto_y = model_x_onto_y.fit()
results_x_onto_y.summary2()

<class 'statsmodels.iolib.summary2.Summary'>
"""
                 Results: Ordinary least squares
=================================================================
Model:              OLS              Adj. R-squared:     0.796   
Dependent Variable: y                AIC:                101.7824
Date:               2017-11-09 17:34 BIC:                104.3875
No. Observations:   100              Log-Likelihood:     -49.891 
Df Model:           1                F-statistic:        391.7   
Df Residuals:       99               Prob (F-statistic): 3.46e-36
R-squared:          0.798            Scale:              0.16041 
--------------------------------------------------------------------
         Coef.     Std.Err.       t       P>|t|     [0.025    0.975]
--------------------------------------------------------------------
x1       0.3789      0.0191    19.7918    0.0000    0.3409    0.4169
-----------------------------------------------------------------
Omnibus:              0.476        Durbin-Watson:           2.166
Prob(Omnibus):        0.788        Jarque-Bera (JB):        0.631
Skew:                 0.115        Prob(JB):                0.729
Kurtosis:             2.685        Condition No.:           1    
=================================================================

"""

Coefficient estimate for ß is 0.3789 with a standard error of 0.0191. That gives a t-statistic of 19.7918 and a p-value of less than 0.0001 associated with the null hypothesis H₀ : ß = 0, so we can reject the null hypothesis.

The coefficient estimates and the associated standard errors are different for y onto x versus x onto y but the t-statistic and p-value are the same.